--- a/src/Internal/Tokenizer/Tokenizer.php
+++ b/src/Internal/Tokenizer/Tokenizer.php
@@ -118,10 +118,11 @@ class Tokenizer implements TokenizerInterface

     private function doTokenize(string $string, ?string $language, ?int $maxTokens = null): TokenCollection
     {
-        if (!isset($this->tokenizers[$language])) {
-            $this->tokenizers[$language] = new LoupeMatcherTokenizer($language);
+        $languageKey = $language ?? '';
+        if (!isset($this->tokenizers[$languageKey])) {
+            $this->tokenizers[$languageKey] = new LoupeMatcherTokenizer($language);
         }

-        $tokenCollection = $this->tokenizers[$language]->tokenize($string, $maxTokens);
+        $tokenCollection = $this->tokenizers[$languageKey]->tokenize($string, $maxTokens);
         $tokenCollectionWithVariants = new TokenCollection();

         foreach ($tokenCollection->all() as $token) {
